### ISSUE TITLE
how to install keepassxc? ubuntu 16.04.

### ISSUE THREAD

**Issue by marcopolo88** [keepassxreboot/keepassxc#1417]
how to install keepassxc? ubuntu 16.04.
followed exactly like described in blog
quick start not possible

opipcp@orangepipcplus:/emmc1$ sudo add-apt-repository ppa:phoerious/keepassxc
 KeePassXC Password Manager (official upstream PPA)
 More info: https://launchpad.net/~phoerious/+archive/ubuntu/keepassxc
Press [ENTER] to continue or ctrl-c to cancel adding it

gpg: keyring `/tmp/tmp48l84yn0/secring.gpg' created
gpg: keyring `/tmp/tmp48l84yn0/pubring.gpg' created
gpg: requesting key 0068FCD6 from hkp server keyserver.ubuntu.com
gpg: /tmp/tmp48l84yn0/trustdb.gpg: trustdb created
gpg: key 0068FCD6: public key "Launchpad PPA for Janek Bevendorff" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
OK
opipcp@orangepipcplus:/emmc1$ apt update
Reading package lists... Done
W: chmod 0700 of directory /var/lib/apt/lists/partial failed - SetupAPTPartialDirectory (1: Operation not permitted)
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
E: Unable to lock directory /var/lib/apt/lists/
W: Problem unlinking the file /var/cache/apt/pkgcache.bin - RemoveCaches (13: Permission denied)
W: Problem unlinking the file /var/cache/apt/srcpkgcache.bin - RemoveCaches (13: Permission denied)
opipcp@orangepipcplus:/emmc1$ sudo apt update
Hit:1 http://ports.ubuntu.com xenial InRelease
Hit:2 http://ppa.launchpad.net/phoerious/keepassxc/ubuntu xenial InRelease
Hit:3 http://ports.ubuntu.com xenial-security InRelease                        
Hit:4 http://ports.ubuntu.com xenial-updates InRelease                         
Hit:5 http://ports.ubuntu.com xenial-backports InRelease                       
Hit:6 http://apt.armbian.com xenial InRelease                                  
Reading package lists... Done                     
Building dependency tree       
Reading state information... Done
80 packages can be upgraded. Run 'apt list --upgradable' to see them.
opipcp@orangepipcplus:/emmc1$ sudo apt install keepassxc
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package keepassxc

opipcp@orangepipcplus:/emmc1$ cat /proc/version
Linux version 3.4.113-sun8i (root@armbian) (gcc version 5.4.0 20160609 (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.4) ) #4 SMP PREEMPT Wed Nov 22 13:45:28 CET 2017

**Comment by phoerious** [comment]
You need to run apt update with sudo as well.

**Comment by seatedscribe** [comment]
He did. Look closer :)

I had the same problem trying to install the new hashing function winner of
the contest bla bla (cannot remember name now) after having added your
PAPA. At the end I just compiled from their git repository, still it's
something happened not only to me it seems...

**Comment by marcopolo88** [comment]
thx @seatedscribe
later also apt-get upgrade, with same picture after
i am tired of the need to compile everything
no quick start here apparently, not as website suggested
keepassx then:

opipcp@orangepipcplus:~$ sudo apt install keepassx
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libllvm4.0
Use 'sudo apt autoremove' to remove it.
The following additional packages will be installed:
  libaudio2 libmng2 libqt4-declarative libqt4-network libqt4-script libqt4-sql libqt4-xml
  libqt4-xmlpatterns libqtcore4 libqtdbus4 libqtgui4 qtcore4-l10n
Suggested packages:
  nas libqt4-declarative-folderlistmodel libqt4-declarative-gestures
  libqt4-declarative-particles libqt4-declarative-shaders qt4-qmlviewer libqt4-dev
  qt4-qtconfig
Recommended packages:
  libqt4-sql-mysql | libqt4-sql-odbc | libqt4-sql-psql | libqt4-sql-sqlite qt-at-spi
The following NEW packages will be installed:
  keepassx libaudio2 libmng2 libqt4-declarative libqt4-network libqt4-script libqt4-sql
  libqt4-xml libqt4-xmlpatterns libqtcore4 libqtdbus4 libqtgui4 qtcore4-l10n
0 upgraded, 13 newly installed, 0 to remove and 0 not upgraded.
Need to get 9,253 kB of archives.
After this operation, 28.9 MB of additional disk space will be used.
Do you want to continue? [Y/n]

**Comment by phoerious** [comment]
> He did. Look closer :)

I see. He ran it twice.

I just tested it in a 16.04 Docker container, installs fine for me. Can you check what

```
cat /etc/apt/sources.list.d/phoerious-*.list
```

gives?

**Comment by seatedscribe** [comment]
Sure, here it is. Could you also reopen this issue please?

```
keepassxc # cat /etc/apt/sources.list.d/phoerious-*.list
deb http://ppa.launchpad.net/phoerious/keepassxc/ubuntu xenial main
deb-src http://ppa.launchpad.net/phoerious/keepassxc/ubuntu xenial main
```

... I still have to solve a 
`error: ‘GCRY_CIPHER_CHACHA20’ was not declared in this scope
`
 even after having installed libgcrypt20-18 and having made sure CMAKE include path is set as per this guide
[https://github.com/keepassxreboot/keepassxc/wiki/Set-up-Build-Environment-on-Linux](https://github.com/keepassxreboot/keepassxc/wiki/Set-up-Build-Environment-on-Linux)

**Comment by phoerious** [comment]
You probably need to delete your CMake cache. The error means, it's still using your old gcryp version.

**Comment by seatedscribe** [comment]
Yes, deleting CMake cache settled it, thanks.

However I noticed that after installing the program and launching from the OS menu (bottom left popup) it cannot find the library (which makes sense since it's a custom path).

So I would suggest to add a step in the wiki guide (Set-up-Build-Environment-on-Linux) after this one:

> Ubuntu 14.04 and 16.04 require newer versions of libgcrypt20-dev, which can be installed from our Ubuntu PPA. To avoid clashes with the upstream libgcrypt20-dev package, our package is named libgcrypt20-18-dev and installs to /opt/libgcrypt20-18. You therefore need to set the following environment variables for CMake to find the required libraries:[...]

**------ start of wiki code ------**

You also need to update library bindings in order for the program to start correctly OS-wide:

**[as root]**
- `echo "/opt/libgcrypt20-18/lib/x86_64-linux-gnu" > /etc/ld.so.conf.d/gcrypt20-18.conf`
- `ldconfig`
- `ldconfig -p |grep libgcrypt` should give something like this:
	libgcrypt.so.20 (libc6,x86-64) => /opt/libgcrypt20-18/lib/x86_64-linux-gnu/libgcrypt.so.20
	libgcrypt.so.20 (libc6,x86-64) => /lib/x86_64-linux-gnu/libgcrypt.so.20
	libgcrypt.so (libc6,x86-64) => /opt/libgcrypt20-18/lib/x86_64-linux-gnu/libgcrypt.so
	libgcrypt.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libgcrypt.so

**------ end of wiki code ------**

Last point - argon2
it seems that using the PPA doesn't always get to the expected result (I had to clone the project to make it working). @phoerious did you have any chance to troubleshoot this, by means of my phoerious-*.list provided above?

@marcopolo88 any update on your attempt to compile KXC? We would love to see it working on your PC!
